README.md

```diff
@@ -1,58 +1,161 @@
+
+### Defining build parameters
 
-2. Prepare configuration
+The [options.yml](blob/master/config/options.yml) points to the appropriate repos and tag/branches.
+
+A typical `options.yml` will only override the source and the refs. This example uses branch `feature1` from kbrock's github repo:
+
+```sh
+mkdir OPTIONS
+vi OPTIONS/options.yml
+```
+
+```yaml
+---
+product_name:      manageiq
+repos:
+  ref:             master
+  manageiq:
+    url:           https://github.com/kbrock/manageiq.git
+    ref:           feature1
+  manageiq_appliance:
+    url:           https://github.com/kbrock/manageiq-appliance.git
+    ref:           feature1
+```
+
+The option file is brought into docker by mounting the directory with `-v $(pwd)/OPTIONS:/root/OPTIONS`
+
+Alternatively, a static directory may used like `/opt/OPTIONS/options.yml`.
+
+### Using a custom NPM registry
+
+Any of the keys in [options.yml](blob/master/config/options.yml) can be overridden.
 
-   - Create a local directory with an `options.yml` with the configuration changes
-     you want to make, then when running the container image, mount that directory
-     with `-v <dir>:/root/OPTIONS`.
    - If overriding the NPM registry, set the `npm_registry` key in the `options.yml`.
    - If building RPMs in Copr,
      - set the `rpm.repo_name` key in the `options.yml`.
      - run the container image with `-v <copr token file>:/root/.config/copr`.
    - If updating the RPM repo it would be helpful to attach a volume to hold the RPM cache with `-v <dir>:/root/rpm_cache`.
      Any RPMs not in the cache will be downloaded to the cache first.
 
-3. Start the container image
+### Artifacts
+
+The purpose of this process is to build code from `manageiq-core`, `manageiq-gemset`, and `manageiq-appliance` into 2 sets of artifacts:
 
-   `docker run -it <image>`
+- `/root/BUILD/rpm_spec/*.tar.gz`
+- `/root/BUILD/rpms/x86_64/manageis-*.rpm`
 
-4. Run the script
+Depending upon your usecase, there are a few ways to obtain these artifacts:
 
-   - Run `bin/build.rb`, optionally overriding git ref (branch or tag) and release type
+- Have the container put the artifacts into a mounted volume.
+- Manually copy the artifacts to a mounted volume.
+- Extracting the artifacts from the docker container with a docker command.
 
-     ```
-     bin/build.rb --git-ref jansa --update-rpm-repo
-     ```
-     ```
-     bin/build.rb --build-type release --git-ref jansa-1-alpha1 --update-rpm-repo
-     ```
+### Having the container build files into a mounted volume
 
-Alternatively, the build can be started directly by passing `build` to the
-`docker run` command:
+Since the files are not uploaded, they need to be retrieved out of the repo before the container exits.
+
+Note: all mounted volumes (i.e.: using the `-v` option) need to be defined in docker's preferences:
 
 ```
-docker run -it <miq-rpm_build image> build [--build-type <type>] [--git-ref <ref>] [--update-rpm-repo]
+Settings > Resources > FILE SHARING
```

Comment:
We probably should mark this as being Docker on Mac specific...this steps doesn't make sense for normal Docker.  I'm ok with a follow-up PR though.

Reply:
thanks - I'll dump this section. now that you found out that you do not need to put every directory into there, this makes the use case much easier